src/main/java/com/urbanairship/datacube/idservices/CachingIdService.java

```diff
@@ -4,104 +4,85 @@
 
 package com.urbanairship.datacube.idservices;
 
-import java.io.IOException;
-import java.util.concurrent.ExecutionException;
-
-import com.yammer.metrics.Metrics;
-import com.yammer.metrics.core.Gauge;
-import org.apache.commons.codec.binary.Hex;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
+import com.google.common.base.Optional;
+import com.google.common.cache.Cache;
```

Comment:
LoadingCache made sense when the semantics were always getOrCreate, now that there's a distinction it's simpler to use a dumb cache and make the db reads explicit.